server/src/stateManagement.mjs

```diff
@@ -700,17 +700,39 @@ function setMethodError(userId, methodName, code, message) {
   });
 }
 
+/* if "group" keyword is passed for scope inside ctx functions, 
+and if userid invoking the YAML contains group, the group of the userId will be used as the "scope" for that call
+else,  the full userid is used along with warning
+*/
+function setScopeForGroupKeyword(userId, scope) {
+  let parseUserId = parseUser(userId)
+  if (parseUserId.group == undefined || parseUserId.group == null) {
+    logger.warning("WARNING: userId does not contain a group. Using full userId for manipulating scratch")
+    scope = userId
+  } else {
+    scope = '~' + parseUserId.group
```

Comment:
I don't know where else "special string construction or parsing" is done for full userId values, but it might be nice to create some kind of "createUserIdForGroup" function for this (alongside the parseUser utility function).